### PR TITLE
SilverStripe 4 compatibility

### PR DESCRIPTION
**Pull request by robbieaverill**
## Summary

* Namespacing added and implemented
* Minor API changes updated from the framework update
* PSR-4 autoloading added to composer.json
* Docs updated
* `.upgade.yml` and legacy class name map added
* Travis configuration updated
* Some changes for PSR-2 style guidelines
* Replaces multivaluefield with a textarea - it seemed like overkill to require the module just for the Job constructor arguments

### CODE REVIEW

**Comment by robbieaverill** [comment]
- [ ] I'll need to investigate and fix the Postgres support

**Comment by mikenz** [comment]
I've just merged this and sending it through our build and test process. I'll let you know if anything comes out of that.

File: README.md [hunk 1]
```diff
@@ -13,14 +13,15 @@ Marcus Nyeholt
 
 ## Requirements
 
-* SilverStripe 3.x
+* SilverStripe 4.x
 * Access to create cron jobs
-* https://github.com/nyeholt/silverstripe-multivaluefield
+* https://github.com/silverstripe-australia/silverstripe-multivaluefield
```
Comment (robbieaverill) [inline README.md]: Thanks, I'll update it tomorrow morning

**Comment by mikenz** [comment]
No obvious issues have shown up here :-)

**Comment by robbieaverill** [comment]
Great! It won't only intended for alpha level anyway, but good to know its stable ish

**Comment by mikenz** [comment]
I haven't found any issues with it.

**Comment by robbieaverill** [comment]
@nyeholt Up to you guys. I didn't come across any problems while testing, but didn't test with Doorman etc. I've just resolved the conflict. If nothing else, it's a good starting point for 4.x compat :)

**Comment by robbieaverill** [comment]
Hey @mikenz @nyeholt - is this there anything else you need for this PR to get over the line?

**Comment by mikenz** [comment]
It's good to go. Our testers haven't found any issues with it, nor has our CI system.

**Comment by nyeholt** [comment]
Okay, in as the master codeline now!

**Pull request merged**